PathPlanning/TimeBasedPathPlanning/GridWithDynamicObstacles.py

```diff
@@ -33,13 +33,24 @@ def __sub__(self, other):
     def __hash__(self):
         return hash((self.x, self.y))
 
+@dataclass
+class Interval:
+    start_time: int
+    end_time: int
 
 class ObstacleArrangement(Enum):
     # Random obstacle positions and movements
     RANDOM = 0
     # Obstacles start in a line in y at center of grid and move side-to-side in x
     ARRANGEMENT1 = 1
 
+"""
+Generates a 2d numpy array with lists for elements.
+"""
+def empty_2d_array_of_lists(x: int, y: int) -> np.ndarray:
+    arr = np.empty((x, y), dtype=object)
+    arr[:] = [[[] for _ in range(y)] for _ in range(x)]
+    return arr
```

Comment:
I think this is equivalent:
```
def empty_2d_array_of_lists(x: int, y: int) -> np.ndarray:
    return np.full((x, y), [], dtype=object)
```

Reply:
I don't think this will work - np.full sets each element to be a reference to the same object. You need to assign each element individually to get unique instances at each element

Reply:
Ah, OK. Sorry.